src/main/java/digital/slovensko/autogram/core/AsicContainer.java

```diff
@@ -0,0 +1,100 @@
+package digital.slovensko.autogram.core;
+
+import digital.slovensko.autogram.core.errors.OriginalDocumentNotFoundException;
+import eu.europa.esig.dss.asic.xades.ASiCWithXAdESContainerExtractor;
+import eu.europa.esig.dss.enumerations.MimeType;
+import eu.europa.esig.dss.enumerations.MimeTypeEnum;
+import eu.europa.esig.dss.model.DSSDocument;
+import eu.europa.esig.dss.validation.CommonCertificateVerifier;
+import eu.europa.esig.dss.validation.SignedDocumentValidator;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+
+public class AsicContainer {
+
+    private DSSDocument asice;
+
+    public AsicContainer(DSSDocument asice) {
+        this.asice = asice;
+    }
+
+    public DSSDocument getOriginalDocument() {
+        var documentValidator = SignedDocumentValidator.fromDocument(asice);
+        documentValidator.setCertificateVerifier(new CommonCertificateVerifier());
+        var signatures = documentValidator.getSignatures();
+        if (signatures.isEmpty()) {
+            throw new OriginalDocumentNotFoundException("No signatures in document");
+        }
+        var advancedSignature = signatures.get(0);
+        var originalDocuments = documentValidator.getOriginalDocuments(advancedSignature.getId());
```

Comment:
Toto je nejaka velmi zvlastna logika, co to robi? Preco tu vlastne sa hrame s podpismi? A preco vlastne len s 1?

Reply:
Po diskusii s @celuchmarek, treba tu dat test a hodit exception ak nie su splnene nase predpoklady - teda, ze je tam len jeden dokument. (Zatial ine asice nepodporujeme)

Reply:
pridal som na to check aj test. Zmenil som trochu to vyberanie priamo cez signedDocuments. A teda táto logika teraz zaručuje, že je v asicu práve jeden a podpísaný dokument.

signedDocuments sú všetky dokumenty na podpísanie v asicu. Nie sú to len tie, ktoré majú k sebe nejaký podpis. Ten test, čo som pridal, má 1 podpísaný a 1 nepodpísaný a vráti počet 2.